_docs/geyser/geyser-api.md

```diff
@@ -0,0 +1,176 @@
+- [Lifecycle](https://github.com/GeyserMC/Geyser/tree/master/api/src/main/java/org/geysermc/geyser/api/event/lifecycle): Events that are sent during Geyser's lifecycle, such as the loading of custom items, resource packs, or Geyser commands.
+
+To see all the events in the respective categories, click on the links above.
+
+##### Examples:
+
+Each method that you want to subscribe to an event needs to be annotated with the @Subscribe annotation (from the GeyserMC events package).
+```java
+@Subscribe
+public void onGeyserLoadResourcePacksEvent(GeyserLoadResourcePacksEvent event) {
+        logger().info("Loading: " + event.resourcePacks().size() + " resource packs.");
+        // you could add a resource pack with event.resourcePacks().add(path-to-pack)
+}
+```
+If you wish to listen to events in a Spigot/Paper plugin or a Fabric mod, you need to register the Geyser Event Bus as a listener first.
+Extensions do not need to do that - they are automatically registered, so a simple @Subscribe annotation is enough.
+
+**Fabric mod example:**
+```java
+public class ExampleMod implements ModInitializer, EventRegistrar {
+    public static final Logger LOGGER = LoggerFactory.getLogger("modid");
+    
+    @Override 
+    public void onInitialize() {
+        ServerLifecycleEvents.SERVER_STARTING.register((server) -> {
+            GeyserApi.api().eventBus().register(this, ExampleMod.class); // register your mod & this class as a listener
+        });
+        
+        LOGGER.info("Geyser is cool!");
+    }
+    
+    // here an event, we subscribe as usual with the @Subscribe annotation
+    @Subscribe 
+    public void onGeyserPostInitializeEvent(GeyserPostInitializeEvent eventad {
+        LOGGER.info("Geyser started!");
+    }
+}
+```
+Do note: We cannot directly register the event but in the mod initializer, since the Geyser API would not be loaded yet.
+Therefore, we register it in the server starting event provided by the Fabric API.
+
+**Spigot/Paper plugin example:**
+
+1. In your plugin.yml, add the following lines:
+```yaml
+  depend: ["Geyser-Spigot"]
+```
+This ensures that your plugin loads after Geyser has, so the Geyser API would be available.
+
+2. In your main class, implement the EventRegistrar interface and register the event bus in the onEnable method:
+```java
+public class ExamplePlugin extends JavaPlugin implements EventRegistrar {
+    
+    @Override
+    public void onEnable(){
+        getLogger().info("Registering Geyser event bus!");
+        GeyserApi.api().eventBus().register(this, ExamplePlugin.class); // register your plugin & this class as a listener
+    }
+
+    // here an event, we subscribe as usual with the @Subscribe annotation
+    @Subscribe
+    public void onGeyserPostInitializeEvent(GeyserPostInitializeEvent event) {
+        getLogger().info("Geyser started!");
+    }
+}
+```
+3. In case the '@Subscribe' annotation didn't work, you can also manually subscribe your method to the event bus:
```

Comment:
```suggestion
3. If you want to provide your event with a consumer, rather than annotating it, you can also manually subscribe your method to the event bus:
```